validation/open_boundaries/cylinder.jl

```diff
@@ -1,127 +1,214 @@
-# This validation script shows open boundaries working in a simple case where the
-# flow remains largely unidirectional and so at one end we have no matching scheme
-# but just prescribe the inflow. At the other end we then make no assumptions about
-# the flow and use a very simple open boundary condition to permit information to 
-# exit the domain. If, for example, the flow at the prescribed boundary was reversed
-# then the model would likely fail.
+using Oceananigans
+using Oceananigans.Models.NonhydrostaticModels: ConjugateGradientPoissonSolver
+using Oceananigans.Solvers: DiagonallyDominantPreconditioner
+using Oceananigans.Operators: ℑxyᶠᶜᵃ, ℑxyᶜᶠᵃ
+using Oceananigans.Solvers: FFTBasedPoissonSolver
+using Printf
+using CUDA
 
-using Oceananigans, CairoMakie
-using Oceananigans.BoundaryConditions: FlatExtrapolationOpenBoundaryCondition
+using Oceananigans.BoundaryConditions: PerturbationAdvectionOpenBoundaryCondition
```

Comment:
Can we keep testing FlatExtrapolation also? Ideally we'd have a do loop and test both, and keep adding to it the more open boundary conditions we implement.